src/PelIfd.php

```diff
@@ -345,7 +345,12 @@ public function load(PelDataWindow $d, $offset)
 
                     if ($starting_offset != $o) {
                         $this->sub[$type] = new PelIfd($type);
-                        $this->sub[$type]->load($d, $o);
+                        try {
+                            $this->sub[$type]->load($d, $o);
+                        } catch (PelDataWindowOffsetException $e) {
+                            unset($this->sub[$type]);
```

Comment:
If you "unset" the array element a hole will be kept in the array. It could be a good idea to assign the new PelIfd object to a temporary object and insert it into the array only on access instead of removing it from the array.